File: code/community/1306408209740730371/whisper-3-nova-3-deepgram-enhancements.cs
using System;
using System.IO;
using System.Net.Http;
using System.Threading.Tasks;

class Program
{
    private static readonly HttpClient client = new HttpClient();

    static async Task Main(string[] args)
    {
        string deepgramApiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        string audioFilePath = "your_audio_file_path.wav";

        if (string.IsNullOrEmpty(deepgramApiKey))
        {
            Console.WriteLine("DEEPGRAM_API_KEY environment variable must be set.");
            return;
        }

        byte[] audioData = File.ReadAllBytes(audioFilePath);
        using (var content = new ByteArrayContent(audioData))
        {
            content.Headers.Add("Authorization", $"Token {deepgramApiKey}");
            content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("audio/wav");

            HttpResponseMessage response = await client.PostAsync("https://api.deepgram.com/v1/listen", content);
            if (response.IsSuccessStatusCode)
            {
                string responseBody = await response.Content.ReadAsStringAsync();
                Console.WriteLine("Transcription results: " + responseBody);
            }
            else
            {
                Console.WriteLine($"Error: {response.StatusCode} - {response.ReasonPhrase}");
            }
        }
    }
}

File: code/community/1306408209740730371/whisper-3-nova-3-deepgram-enhancements.go
package main

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
)

func main() {
	apiKey := os.Getenv("DEEPGRAM_API_KEY")
	if apiKey == "" {
		fmt.Println("DEEPGRAM_API_KEY environment variable must be set.")
		return
	}

	audioFilePath := "your_audio_file_path.wav"
	audioData, err := ioutil.ReadFile(audioFilePath)
	if err != nil {
		fmt.Println("Error reading audio file:", err)
		return
	}

	url := "https://api.deepgram.com/v1/listen"
	req, err := http.NewRequest("POST", url, bytes.NewBuffer(audioData))
	if err != nil {
		fmt.Println("Error creating request:", err)
		return
	}

	req.Header.Set("Authorization", "Token "+apiKey)
	req.Header.Set("Content-Type", "audio/wav")

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Println("Error making request:", err)
		return
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println("Error reading response body:", err)
		return
	}

	if resp.StatusCode == http.StatusOK {
		fmt.Println("Transcription results:", string(body))
	} else {
		fmt.Printf("Error: %d - %s\n", resp.StatusCode, string(body))
	}
}